pkgs/by-name/co/collabora-online/package.nix

```diff
@@ -0,0 +1,121 @@
+{
+  autoreconfHook,
+  cairo,
+  cppunit,
+  fetchFromGitHub,
+  fetchNpmDeps,
+  lib,
+  libcap,
+  libpng,
+  libreoffice-collabora,
+  nodejs,
+  npmHooks,
+  pam,
+  pango,
+  pixman,
+  pkg-config,
+  poco,
+  python3,
+  rsync,
+  stdenv,
+  zstd,
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "collabora-online";
+  version = "24.04.5-3";
+
+  src = fetchFromGitHub {
+    owner = "CollaboraOnline";
+    repo = "online";
+    rev = "refs/tags/cp-${finalAttrs.version}";
+    hash = "sha256-wy+vfRmKYULq2kg0MKhdxTD7HC63cTd1cYX5xN7LYgg=";
+  };
+
+  nativeBuildInputs = [
+    autoreconfHook
+    libcap
+    nodejs
+    npmHooks.npmConfigHook
+    pkg-config
+    python3
+    python3.pkgs.lxml
+    python3.pkgs.polib
+    rsync
+  ];
+
+  buildInputs = [
+    cairo
+    cppunit
+    libpng
+    pam
+    pango
+    pixman
+    poco
+    zstd
+  ];
+
+  configureFlags = [
+    "--disable-setcap"
+    "--disable-werror"
+    "--enable-silent-rules"
+    "--with-lo-path=${libreoffice-collabora}/lib/collaboraoffice"
+    "--with-lokit-path=${libreoffice-collabora.src}/include"
+  ];
+
+  env = {
+    SETCAP = "${libcap}/bin/setcap";
+  };
+
+  postPatch =
+    ''
+      cp ${./package-lock.json} ${finalAttrs.npmRoot}/package-lock.json
+      patchShebangs .
+      substituteInPlace configure.ac --replace-fail '/usr/bin/env python3' python3
+      substituteInPlace coolwsd-systemplate-setup --replace-fail /bin/pwd pwd
+    ''
+    # Use setcap'd wrappers from the wrappers dir, not from the "application path".
+    + ''
+      substituteInPlace common/JailUtil.cpp --replace-fail \
+        'Poco::Path(Util::getApplicationPath(), "coolmount").toString()' \
+        'std::string("/run/wrappers/bin/coolmount")'
+      substituteInPlace wsd/COOLWSD.cpp --replace-fail \
+        ' forKitPath += "coolforkit";' \
+        ' forKitPath = "/run/wrappers/bin/coolforkit";'
```

Comment:
If we take them from PATH, then the package easier works on other distros

Reply:
The code assumes absolute paths. I could come up with a patch that calls `which coolforkit` from C++ code, but it will be not exactly a trivial patch. Are we okay with such patches?

Or maybe something like accepting a `NIX_COLLABORA_COOLFORKIT` env var. Do we have anything like these in nixpkgs?

Reply:
If the code wants absolute paths, then we hardcode them for now.

Changing c/++ code is always open for memory issues and it is usually good to avoid that if possible.

We have such envs in nixpkgs but only if it needs to be dynamic. Lets start simple here first.

Reply:
Removed these fixups entirely as they are not required with `mount_namespaces` option.